core/src/com/unciv/models/ruleset/unique/UniqueType.kt

```diff
@@ -806,6 +807,8 @@ enum class UniqueType(
     OneTimeFreeBelief("Gain a free [beliefType] belief", UniqueTarget.Triggerable),
     OneTimeTriggerVoting("Triggers voting for the Diplomatic Victory", UniqueTarget.Triggerable),  // used in Building
 
+    OneTimeGainStockpile("Instantly gain [amount] [stockpile]", UniqueTarget.Triggerable, flags = setOf(UniqueFlag.AcceptsSpeedModifier)),
+    OneTimeGainStockpileRange("Instantly gain [amount]-[amount] [stockpile]", UniqueTarget.Triggerable, flags = setOf(UniqueFlag.AcceptsSpeedModifier)),
```

Comment:
Should use existing Instantly Consumes / Instantly Provides uniques

Reply:
I kinda have issues with those uniques, tbh. Basically \*nothing\* else uses that design style for uniques except for the non stockpile resource uniques. For the non stockpile resource uniques, there's a mechanical reasoning behind why it's 2 separate uniques (there's various UI considerations such as being how we automatically place a restriction of where stuff could be built). For stockpiles, there is no mechanical reasoning for having 2 separate uniques

I mean, just look at nearly all other use cases and I'd argue that it's those that need deprecating here, rather than this using those uniques

Reply:
Not sure what you mean? Do you mean using "instantly gain [-2]" instead of "instantly lose [2]"?

Reply:
Yes